File: Easy/Bird and maximum fruit gathering.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;
#define ll long long

// } Driver Code Ends

class Solution
{
    public:
    int maxFruits(int arr[], int n, int m)
    {
        int somma = 0;
        
        for (int i=0; i<m; i++) somma += arr[i];
    
        int ans = somma;
        
        if (m >= n) return somma;
        
        int fine = m;
        
        for (int i=0; i<n; i++)
        {
            somma -= arr[i];
            somma += arr[fine];
            
            ans = max(ans, somma);
            
            fine = (fine + 1) %n;
        }
        
        
        return ans;
    }
    
};

//{ Driver Code Starts.
int main() 
{
    int t,i,n,m,j;
    cin>>t;
    while(t--){
        cin>>n>>m;
        if(m>n)m=n;
        int a[n],sum=0,maxSum=0;
        for(i=0;i<n;i++)cin>>a[i];
        Solution ob;
        cout<<ob.maxFruits(a,n,m)<<"\n";
    }
	return 0;
}
// } Driver Code Ends

File: Easy/Bird and maximum fruit gathering.py
#User function Template for python3

class Solution:
    def maxFruits(self, arr, n, m):
        head = []
        for i in range(m):
            head += [arr[i]]
        
        tail = []
        for i in range(n-m, n):
            tail += [arr[i]]
        
        arr = tail + arr + head
        
        #print(arr)
        
        ans = -1
        temp = 0
        
        for i in range(n + (m * 2)):
            temp += arr[i]
            
            if i >= m:
                temp -= arr[i-m]
                ans = max(ans, temp)
        
            #print(temp)
            
        return ans
                
            


#{ 
 # Driver Code Starts
#Initial Template for Python 3

t=int(input())
for _ in range(0,t):
    #n=int(input())
    l=list(map(int,input().split()))
    n=l[0]
    m=l[1]
    a = list(map(int,input().split()))
    ob=Solution()
    ans=ob.maxFruits(a,n,m)
    print(ans)
# } Driver Code Ends